services/dataAccessLayer.js

```diff
@@ -34,6 +46,22 @@ const retreiveFilteredUsers = async (query) => {
   return users;
 };
 
+const retrieveMembers = async (query) => {
+  const allUsers = await members.fetchUsers(query);
+  allUsers.forEach((element) => {
+    removeSensitiveInfo(element);
+  });
+  return allUsers;
+};
+
+const retrieveUsersWithRole = async (role) => {
+  const users = await members.fetchUsersWithRole(role);
+  users.forEach((element) => {
+    removeSensitiveInfo(element);
+  });
+  return users;
+};
+
 const removeSensitiveInfo = function (obj) {
```

Comment:
removeSensitiveInfo() function uses the in operator to check if a property exists on an object. However, the in operator is not safe against object injection attacks. This means that an attacker could inject a malicious object into the obj parameter, and the removeSensitiveInfo() function would not be able to detect it.

To fix this bug, you can use the hasOwnProperty() method instead of the in operator. The hasOwnProperty() method only returns true if the property exists on the object and was not inherited from a parent object. This makes it safe against object injection attacks.

Reply:
But even after this update the warning is still visible @mailmesriza98 @iamitprakash 
![image](https://github.com/Real-Dev-Squad/website-backend/assets/97341921/29240193-309d-4760-a5b4-73b7c5e6fa23)